service/datasets/create_example_dataset.py

```diff
@@ -0,0 +1,20 @@
+import random
+
+"""
+Code to create an example dataset for testing. c is y-value; 
+c is 0 if a is low and is high, and is 1 otherwise
+"""
+
+list_a = [random.randint(0, 50) for i in range(1000)]
+list_b = [random.randint(0, 50) for i in range(1000)]
+f = open("example.csv", "w", newline='')

```

Comment:
Switching the f = + f.close to 

with open("example.csv", "w", newline='') as f:
     the rest of the code

Helps ensure close is called in the case of a failure. It uses the context manager paradigm in python